homeassistant/components/sipcall/config_flow.py

```diff
@@ -0,0 +1,46 @@
+"""Config flow for SIP Call integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_SIP_DOMAIN, CONF_SIP_SERVER, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_SIP_SERVER): str,
+        vol.Required(CONF_SIP_DOMAIN): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SIP Call."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            entry_title = f"{user_input[CONF_USERNAME]}_{user_input[CONF_SIP_DOMAIN]}"
+            return self.async_create_entry(
+                title=entry_title, data=user_input | {CONF_NAME: entry_title}
```

Comment:
```suggestion
                title=entry_title, data=user_input
```
No need to store the name twice

Reply:
It won't work without this. The notify platform entry needs a name. See also [here](https://github.com/home-assistant/core/blob/dev/homeassistant/components/discord/config_flow.py#L70).

Reply:
I can see that we add it, but I can't see where we use it. Can you point me to that?

Reply:
It is used [here](https://github.com/home-assistant/core/blob/dev/homeassistant/components/notify/legacy.py#L117). It goes through `slugify` to make the service name.

Reply:
Interesting one, I'll try to find some people to discuss this with. Thanks

Reply:
Yeah it's a bit annoying, because if the user changes the `title` of the config entry, it won't change the underlying name of the service to call. Would be nice if the user could change that too

Reply:
I know for a fact that there are some ideas to phase out this type of notify services and introduce some kind of NotifyEntity to call a generic service onto an entity instead of having a service per entity, that wouldve been ideal here

Reply:
Let me know when there's a shiny new notify implementation ;)